pylint/checkers/strings.py

```diff
@@ -672,6 +727,33 @@ def visit_assign(self, node):
         if isinstance(node.value, astroid.Const) and isinstance(node.value.value, str):
             self.check_for_concatenated_strings([node.value], "assignment")
 
+    def check_for_consistent_string_delimiters(
```

Comment:
I don't like the dependency on work having been done in the caller to count delimiters.  The upside I see for that work being done there is that tokens are being iterated through there, already, for a variety of reasons.  What do you folks think about moving that logic into this method for its own pass through all the tokens?

Reply:
If that work is pulled into this method, we can dispense with `self._string_delimiters`, too.  That's a nice bonus.

Reply:
I'm just going to make that change.